File: dipeo/models/src/enums.ts
/**
 * Shared enum definitions for diagram-related types
 * These enums are used by both frontend and backend via code generation
 */

export enum NodeType {
  START = 'start',
  PERSON_JOB = 'person_job',
  CONDITION = 'condition',
  CODE_JOB = 'code_job',
  API_JOB = 'api_job',
  ENDPOINT = 'endpoint',
  DB = 'db',
  USER_RESPONSE = 'user_response',
  NOTION = 'notion',
  PERSON_BATCH_JOB = 'person_batch_job',
  HOOK = 'hook',
  TEMPLATE_JOB = 'template_job',
  JSON_SCHEMA_VALIDATOR = 'json_schema_validator',
  TYPESCRIPT_AST = 'typescript_ast',
  SUB_DIAGRAM = 'sub_diagram'
}

export enum HandleDirection {
  INPUT = 'input',
  OUTPUT = 'output'
}

export enum HandleLabel {
  DEFAULT = 'default',
  FIRST = 'first',
  CONDITION_TRUE = 'condtrue',
  CONDITION_FALSE = 'condfalse',
  SUCCESS = 'success',
  ERROR = 'error',
  RESULTS = 'results'
}

export enum DataType {
  ANY = 'any',
  STRING = 'string',
  NUMBER = 'number',
  BOOLEAN = 'boolean',
  OBJECT = 'object',
  ARRAY = 'array'
}

export enum MemoryView {
  ALL_INVOLVED = 'all_involved',
  SENT_BY_ME = 'sent_by_me',
  SENT_TO_ME = 'sent_to_me',
  SYSTEM_AND_ME = 'system_and_me',
  CONVERSATION_PAIRS = 'conversation_pairs',
  ALL_MESSAGES = 'all_messages',
}

export enum MemoryProfile {
  FULL = 'FULL',
  FOCUSED = 'FOCUSED',
  MINIMAL = 'MINIMAL',
  GOLDFISH = 'GOLDFISH',
  CUSTOM = 'CUSTOM'
}

export enum DiagramFormat {
  NATIVE = 'native',
  LIGHT = 'light',
  READABLE = 'readable'
}

export enum DBBlockSubType {
  FIXED_PROMPT = 'fixed_prompt',
  FILE = 'file',
  CODE = 'code',
  API_TOOL = 'api_tool'
}

export enum ContentType {
  RAW_TEXT = 'raw_text',
  CONVERSATION_STATE = 'conversation_state',
  OBJECT = 'object',
  EMPTY = 'empty',
  GENERIC = 'generic',
  VARIABLE = 'variable'
}

export enum SupportedLanguage {
  PYTHON = 'python',
  TYPESCRIPT = 'typescript',
  BASH = 'bash',
  SHELL = 'shell'
}

export enum HttpMethod {
  GET = 'GET',
  POST = 'POST',
  PUT = 'PUT',
  DELETE = 'DELETE',
  PATCH = 'PATCH'
}

export enum HookType {
  SHELL = 'shell',
  WEBHOOK = 'webhook',
  PYTHON = 'python',
  FILE = 'file'
}

export enum HookTriggerMode {
  NONE = 'none',
  MANUAL = 'manual',
  HOOK = 'hook'
}

export enum ExecutionStatus {
  PENDING = 'PENDING',
  RUNNING = 'RUNNING',
  PAUSED = 'PAUSED',
  COMPLETED = 'COMPLETED',
  FAILED = 'FAILED',
  ABORTED = 'ABORTED',
  SKIPPED = 'SKIPPED'
}

export enum NodeExecutionStatus {
  PENDING = 'PENDING',
  RUNNING = 'RUNNING',
  PAUSED = 'PAUSED',
  COMPLETED = 'COMPLETED',
  FAILED = 'FAILED',
  ABORTED = 'ABORTED',
  SKIPPED = 'SKIPPED',
  MAXITER_REACHED = 'MAXITER_REACHED'
}

export enum EventType {
  EXECUTION_STATUS_CHANGED = 'EXECUTION_STATUS_CHANGED',
  NODE_STATUS_CHANGED = 'NODE_STATUS_CHANGED',
  NODE_PROGRESS = 'NODE_PROGRESS',
  INTERACTIVE_PROMPT = 'INTERACTIVE_PROMPT',
  INTERACTIVE_RESPONSE = 'INTERACTIVE_RESPONSE',
  EXECUTION_ERROR = 'EXECUTION_ERROR',
  EXECUTION_UPDATE = 'EXECUTION_UPDATE'
}

export enum LLMService {
  OPENAI = 'openai',
  ANTHROPIC = 'anthropic',
  GOOGLE = 'google',
  GEMINI = 'gemini',
  BEDROCK = 'bedrock',
  VERTEX = 'vertex',
  DEEPSEEK = 'deepseek'
}

export enum APIServiceType {
  OPENAI = 'openai',
  ANTHROPIC = 'anthropic',
  GOOGLE = 'google',
  GEMINI = 'gemini',
  BEDROCK = 'bedrock',
  VERTEX = 'vertex',
  DEEPSEEK = 'deepseek',
  NOTION = 'notion',
  GOOGLE_SEARCH = 'google_search',
  SLACK = 'slack',
  GITHUB = 'github',
  JIRA = 'jira'
}

export enum NotionOperation {
  CREATE_PAGE = 'create_page',
  UPDATE_PAGE = 'update_page',
  READ_PAGE = 'read_page',
  DELETE_PAGE = 'delete_page',
  CREATE_DATABASE = 'create_database',
  QUERY_DATABASE = 'query_database',
  UPDATE_DATABASE = 'update_database'
}

export enum ToolType {
  WEB_SEARCH = 'web_search',
  WEB_SEARCH_PREVIEW = 'web_search_preview',
  IMAGE_GENERATION = 'image_generation'
}

export enum ToolSelection {
  NONE = 'none',
  IMAGE = 'image', 
  WEBSEARCH = 'websearch'
}

File: dipeo/diagram_generated/enums.py
"""
Generated enum definitions for DiPeO.
Avoid editing THIS FILE DIRECTLY.
Generated at: 2025-10-04T16:50:38.989298
"""

from enum import Enum


class DataType(str, Enum):
    """"""
    
    ANY = "any"
    
    STRING = "string"
    
    NUMBER = "number"
    
    BOOLEAN = "boolean"
    
    OBJECT = "object"
    
    ARRAY = "array"
    


class ContentType(str, Enum):
    """"""
    
    RAW_TEXT = "raw_text"
    
    CONVERSATION_STATE = "conversation_state"
    
    OBJECT = "object"
    
    EMPTY = "empty"
    
    GENERIC = "generic"
    
    VARIABLE = "variable"
    
    BINARY = "binary"
    


class HandleDirection(str, Enum):
    """"""
    
    INPUT = "input"
    
    OUTPUT = "output"
    


class HandleLabel(str, Enum):
    """"""
    
    DEFAULT = "default"
    
    FIRST = "first"
    
    CONDTRUE = "condtrue"
    
    CONDFALSE = "condfalse"
    
    SUCCESS = "success"
    
    ERROR = "error"
    
    RESULTS = "results"
    


class DiagramFormat(str, Enum):
    """"""
    
    NATIVE = "native"
    
    LIGHT = "light"
    
    READABLE = "readable"
    


class Status(str, Enum):
    """"""
    
    PENDING = "pending"
    
    RUNNING = "running"
    
    PAUSED = "paused"
    
    COMPLETED = "completed"
    
    FAILED = "failed"
    
    ABORTED = "aborted"
    
    SKIPPED = "skipped"
    
    MAXITER_REACHED = "maxiter_reached"
    


class FlowStatus(str, Enum):
    """"""
    
    WAITING = "waiting"
    
    READY = "ready"
    
    RUNNING = "running"
    
    BLOCKED = "blocked"
    


class CompletionStatus(str, Enum):
    """"""
    
    SUCCESS = "success"
    
    FAILED = "failed"
    
    SKIPPED = "skipped"
    
    MAX_ITER = "max_iter"
    


class ExecutionPhase(str, Enum):
    """"""
    
    MEMORY_SELECTION = "memory_selection"
    
    DIRECT_EXECUTION = "direct_execution"
    
    DECISION_EVALUATION = "decision_evaluation"
    
    DEFAULT = "default"
    


class EventType(str, Enum):
    """"""
    
    EXECUTION_STARTED = "execution_started"
    
    EXECUTION_COMPLETED = "execution_completed"
    
    EXECUTION_ERROR = "execution_error"
    
    NODE_STARTED = "node_started"
    
    NODE_COMPLETED = "node_completed"
    
    NODE_ERROR = "node_error"
    
    NODE_OUTPUT = "node_output"
    
    EXECUTION_LOG = "execution_log"
    
    INTERACTIVE_PROMPT = "interactive_prompt"
    
    INTERACTIVE_RESPONSE = "interactive_response"
    


class LLMService(str, Enum):
    """"""
    
    OPENAI = "openai"
    
    ANTHROPIC = "anthropic"
    
    CLAUDE_CODE = "claude-code"
    
    CLAUDE_CODE_CUSTOM = "claude-code-custom"
    
    GOOGLE = "google"
    
    GEMINI = "gemini"
    
    OLLAMA = "ollama"
    


class APIServiceType(str, Enum):
    """"""
    
    OPENAI = "openai"
    
    ANTHROPIC = "anthropic"
    
    GOOGLE = "google"
    
    GEMINI = "gemini"
    
    OLLAMA = "ollama"
    
    CLAUDE_CODE = "claude-code"
    
    CLAUDE_CODE_CUSTOM = "claude-code-custom"
    


class ToolType(str, Enum):
    """"""
    
    WEB_SEARCH = "web_search"
    
    WEB_SEARCH_PREVIEW = "web_search_preview"
    
    IMAGE_GENERATION = "image_generation"
    


class ToolSelection(str, Enum):
    """"""
    
    NONE = "none"
    
    IMAGE = "image"
    
    WEBSEARCH = "websearch"
    


class AuthType(str, Enum):
    """"""
    
    NONE = "none"
    
    BEARER = "bearer"
    
    BASIC = "basic"
    
    API_KEY = "api_key"
    


class RetryStrategy(str, Enum):
    """"""
    
    NONE = "none"
    
    LINEAR = "linear"
    
    EXPONENTIAL = "exponential"
    
    FIBONACCI = "fibonacci"
    
    CONSTANT = "constant"
    
    EXPONENTIAL_BACKOFF = "exponential_backoff"
    
    LINEAR_BACKOFF = "linear_backoff"
    
    FIXED_DELAY = "fixed_delay"
    


class DBBlockSubType(str, Enum):
    """"""
    
    FIXED_PROMPT = "fixed_prompt"
    
    FILE = "file"
    
    CODE = "code"
    
    API_TOOL = "api_tool"
    


class SupportedLanguage(str, Enum):
    """"""
    
    PYTHON = "python"
    
    TYPESCRIPT = "typescript"
    
    BASH = "bash"
    
    SHELL = "shell"
    


class HttpMethod(str, Enum):
    """"""
    
    GET = "get"
    
    POST = "post"
    
    PUT = "put"
    
    DELETE = "delete"
    
    PATCH = "patch"
    


class HookType(str, Enum):
    """"""
    
    SHELL = "shell"
    
    WEBHOOK = "webhook"
    
    PYTHON = "python"
    
    FILE = "file"
    


class HookTriggerMode(str, Enum):
    """"""
    
    NONE = "none"
    
    MANUAL = "manual"
    
    HOOK = "hook"
    


class ConditionType(str, Enum):
    """"""
    
    DETECT_MAX_ITERATIONS = "detect_max_iterations"
    
    CHECK_NODES_EXECUTED = "check_nodes_executed"
    
    CUSTOM = "custom"
    
    LLM_DECISION = "llm_decision"
    


class TemplateEngine(str, Enum):
    """"""
    
    INTERNAL = "internal"
    
    JINJA2 = "jinja2"
    


class NodeType(str, Enum):
    """"""
    
    START = "start"
    
    PERSON_JOB = "person_job"
    
    CONDITION = "condition"
    
    CODE_JOB = "code_job"
    
    API_JOB = "api_job"
    
    ENDPOINT = "endpoint"
    
    DB = "db"
    
    USER_RESPONSE = "user_response"
    
    HOOK = "hook"
    
    TEMPLATE_JOB = "template_job"
    
    JSON_SCHEMA_VALIDATOR = "json_schema_validator"
    
    TYPESCRIPT_AST = "typescript_ast"
    
    SUB_DIAGRAM = "sub_diagram"
    
    INTEGRATED_API = "integrated_api"
    
    IR_BUILDER = "ir_builder"
    
    DIFF_PATCH = "diff_patch"
    


class Severity(str, Enum):
    """"""
    
    ERROR = "error"
    
    WARNING = "warning"
    
    INFO = "info"
    


class EventPriority(str, Enum):
    """"""
    
    LOW = "low"
    
    NORMAL = "normal"
    
    HIGH = "high"
    
    CRITICAL = "critical"
    


class QueryOperationType(str, Enum):
    """"""
    
    QUERY = "query"
    
    MUTATION = "mutation"
    
    SUBSCRIPTION = "subscription"
    


class CrudOperation(str, Enum):
    """"""
    
    GET = "get"
    
    LIST = "list"
    
    CREATE = "create"
    
    UPDATE = "update"
    
    DELETE = "delete"
    
    SUBSCRIBE = "subscribe"
    


class QueryEntity(str, Enum):
    """"""
    
    DIAGRAM = "Diagram"
    
    PERSON = "Person"
    
    EXECUTION = "Execution"
    
    API_KEY = "ApiKey"
    
    CONVERSATION = "Conversation"
    
    FILE = "File"
    
    NODE = "Node"
    
    PROMPT_TEMPLATE = "PromptTemplate"
    
    SYSTEM = "System"
    


class FieldPreset(str, Enum):
    """"""
    
    MINIMAL = "minimal"
    
    STANDARD = "standard"
    
    DETAILED = "detailed"
    
    FULL = "full"
    


class FieldGroup(str, Enum):
    """"""
    
    METADATA = "metadata"
    
    TIMESTAMPS = "timestamps"
    
    RELATIONSHIPS = "relationships"
    
    CONFIGURATION = "configuration"
    



# Export all enums
__all__ = [
    
    "DataType",
    
    "ContentType",
    
    "HandleDirection",
    
    "HandleLabel",
    
    "DiagramFormat",
    
    "Status",
    
    "FlowStatus",
    
    "CompletionStatus",
    
    "ExecutionPhase",
    
    "EventType",
    
    "LLMService",
    
    "APIServiceType",
    
    "ToolType",
    
    "ToolSelection",
    
    "AuthType",
    
    "RetryStrategy",
    
    "DBBlockSubType",
    
    "SupportedLanguage",
    
    "HttpMethod",
    
    "HookType",
    
    "HookTriggerMode",
    
    "ConditionType",
    
    "TemplateEngine",
    
    "NodeType",
    
    "Severity",
    
    "EventPriority",
    
    "QueryOperationType",
    
    "CrudOperation",
    
    "QueryEntity",
    
    "FieldPreset",
    
    "FieldGroup",
    
]
